helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/model/ClusterModelProvider.java

```diff
@@ -154,39 +329,35 @@ public static ClusterModel generateClusterModelFromCurrentState(
       for (String resourceName : replicaMap.keySet()) {
         Set<AssignableReplica> replicas = replicaMap.get(resourceName);
         // 1. if the resource config/idealstate is changed, need to reassign.
-        // 2. if the resource does appear in the best possible assignment, need to reassign.
+        // 2. if the resource does appear in the current assignment, need to reassign.
```

Comment:
Does or does not appear?

Reply:
`!currentAssignment.containsKey(resourceName))`?

Reply:
Good catch! Fixed.